pkg/template/compiled_template_error.go

```diff
@@ -204,6 +204,26 @@ func (CompiledTemplateMultiError) hintMsg(errMsg string) string {
 		hintMsg = "use 'None' instead of 'none' to indicate no value"
 	case "unhandled index operation struct[string]":
 		hintMsg = "use getattr(...) to access struct field programmatically"
+	case "unknown binary op: bool | bool":
+		hintMsg = "use 'or' instead of '|' to combine boolean expressions"
+	case "unknown binary op: bool & bool":
+		hintMsg = "use 'and' instead of '&' to combine boolean expressions"
+	case "got '&', want primary expression":
+		for _, pos := range err.Positions {
+			codeLine := pos.TemplateLine.Instruction.code
+			if strings.Contains(codeLine, "&&") {
+				hintMsg = "use 'and' instead of '&&' to combine boolean expressions"
+				break
+			}
+		}
```

Comment:
💡 If the body of this case and the subsequent one are genuinely the same logic (and not just incidentally similar), let's factor that into a helper function. This has the double effect of making _this_ function more readable _and_ reducing the total number of codepaths (which generally reduces the risk of bugs).

Just like `hintMsg()`, this new function it should be a method on `CompiledTemplateMultiError` so as to namespace it (and not be defining this as a package-level function).

Reply:
Got it. Made the appropriate changes. Thank you.